pageserver/src/layered_repository/image_layer.rs

```diff
@@ -114,7 +114,7 @@ impl Layer for ImageLayer {
     }
 
     fn get_end_lsn(&self) -> Lsn {
-        self.lsn
+        self.lsn + 1
```

Comment:
Why this change?
In my understanding, `start_lsn == end_lsn` is important invariant of ImageLayer and some functions may rely on that.
I.e. ```start_lsn = prev_layer.get_end_lsn();```

Besides, I don't think it's a good idea to use some non-existing (and not aligned) lsn.

Reply:
So, previously we would create layers like this:
```
a: rel_1663_13990_16389_0_0_000000006CB4F3A0
b: rel_1663_13990_16389_0_0_0000000061E44298_000000006CB4F3A0
c: rel_1663_13990_16389_0_0_0000000061E44298
d: rel_1663_13990_16389_0_0_000000005CE5BAE0_0000000061E44298
```
The start-end LSN ranges of these layers, as reported by the `get_start_lsn()` and `get_end_lsn()` functions were:
```
a: 0/6CB4F3A0 - 0/6CB4F3A0
b: 0/61E44298 - 0/6CB4F3A0
c: 0/61E44298 - 0/61E44298
d: 0/5CE5BAE0 - 0/61E44298
```

There is some ambiguity here. If you're looking for page at LSN `0/6CB4F3A0`, which layer do you look into? There are two layers with that end-LSN: a and b. The correct answer is to look into layer 'b'. 'a' is a delta layer and it contains all the WAL leading up to that LSN, but it doesn't contain the record at that exact LSN, i.e. the end-boundary is exclusive. 'b' is an image layer and contains the state as it is at that LSN, so the end-boundary is inclusive.

This PR clarifies that, so that the end-LSN is always exclusive. So with this PR, the LSN ranges in the above situation would be:
```
a: 0/6CB4F3A0 - 0/6CB4F3A1
b: 0/61E44298 - 0/6CB4F3A0
c: 0/61E44298 - 0/61E44299
d: 0/5CE5BAE0 - 0/61E44298
```
Now it's more clear that you need to look at layer 'a'.

Reply:
I updated the comments and README accordingly